src/optimizer/optimizer.cpp

```diff
@@ -94,27 +94,63 @@ void Optimizer::OptimizeLoop(int root_group_id,
 shared_ptr<planner::AbstractPlan> Optimizer::BuildPelotonPlanTree(
     const unique_ptr<parser::SQLStatementList> &parse_tree_list,
     const std::string default_database_name,
-    concurrency::TransactionContext *txn) {
+    concurrency::TransactionContext *txn, const std::string session_namespace) {
   // Base Case
   if (parse_tree_list->GetStatements().size() == 0) return nullptr;
 
   unique_ptr<planner::AbstractPlan> child_plan = nullptr;
 
   auto parse_tree = parse_tree_list->GetStatements().at(0).get();
 
+  auto stmt_type = parse_tree->GetType();
```

Comment:
It seems like it is possible to merge some branches in the select statement.

Reply:
@HenryZhou0333 What do you mean?